cpp/include/cuvs/neighbors/graph_build_types.hpp

```diff
@@ -27,7 +27,7 @@ namespace cuvs::neighbors {
  * @{
  */
 
-enum GRAPH_BUILD_ALGO { BRUTE_FORCE = 0, IVF_PQ = 1, NN_DESCENT = 1 };
+enum GRAPH_BUILD_ALGO { BRUTE_FORCE = 0, IVF_PQ = 1, NN_DESCENT = 2 };
```

Comment:
We used this pattern of suffixing public API files in RAFT w/ a `_types.hpp` file specifically because it's header only and so including more header files than needed can be costly. That being said, we no longer have that constraint here, since the public API headers are actually very lightweight. For that reason, I would highly suggest we move this directly into `all_neighbors.hpp` or, if it will need to be shared by multiple all_neighbors headers, moving it into `all_neighbors/common.hpp`. I would suggest the former at the present time, though.

Reply:
cagra uses this as well, and that's why it's not under the `all_neighbors` directory. We can't put this into `include/cuvs/neighbors/common.hpp` because that introduces a circular dependency. 

what would be a good way to handle this in this case?

Reply:
If this is being used by more than just `all_neighbors` then I'm fine promiting it to a more common place. My only point is that these `_types` files should probbaly be consolidated into files called `common.hpp` since in cuVS the definitions are very lightweight.

Reply:
Let me put this in a separate PR. Issue added [here](https://github.com/rapidsai/cuvs/issues/1233)